torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -550,13 +550,13 @@ void Reducer::delay_all_reduce() {
   // unused_parameters_ will not change after 1st iteration.
   unused_parameters_.clear();
 
+  require_finalize_ = true;
   // copy all gradients to buckets
   for (const auto variable_index : c10::irange(params_.size())) {
```

Comment:
Just to confirm, `params_` is always non-empty?

Reply:
It should be, there's somewhere in the reducer that raises an error if DDP is created with no params IIUC. Regardless, finalize_backward() sets this to false upon entry of the function so it should be the same.